back/infolica/views/client.py

```diff
@@ -9,6 +9,64 @@
 import json
 from datetime import datetime
 from sqlalchemy import cast, or_, String
+import re
+
+
+def _set_client_aggregated_name(client, sep=', '):
+    nom_ = sep.join(filter(None, [
+        client.entreprise,
+        ' '.join(filter(None, [client.titre, client.prenom, client.nom])),
+        client.co,
+        client.adresse,
+        ' '.join(filter(None, [client.npa, client.localite])),
+        'SAP: ' + (client.no_sap if client.no_sap is not None else '-'),
+        'BDP/BDEE: ' + (client.no_bdp_bdee if client.no_bdp_bdee is not None else '-')
+    ]))
+
+    if client.sortie is not None:
+        nom_ = '(ancien client) ' + nom_
+    
+    return nom_
+
+
+def _multipleAttributesClientSearch(request, searchTerm, old_clients=False, search_limit='default'):
+    if search_limit == 'default':
+        search_limit = int(request.registry.settings['search_limit'])
+
+    searchTerms = re.split(r'\s|\,|\:|\-|SAP|BDP/BDEE', searchTerm.strip())
+    searchTerms = list(filter(None, searchTerms))
+
+    query = request.dbsession.query(Client)
+    if not old_clients:
+        query = query.filter(Client.sortie == None)
+
+    if len(searchTerms) > 0:
+        for term in searchTerms:
+            term = '%' + str(term) + '%'
+            query = query.filter(
```

Comment:
You could create a ts vector field and use this one for search purposes

Reply:
But that's a different story (meaning that it is quite a lot of work...)

Reply:
I would be interested, but for now, I can live with this solution, which is working well 😉